pkg/provisioner/ironic/raid.go

```diff
@@ -172,3 +221,19 @@ func BuildRAIDCleanSteps(raid *metal3v1alpha1.RAIDConfig) (cleanSteps []nodes.Cl
 	)
 	return
 }
+
+func checkRAIDConfigure(raidInterface string, raid *metal3v1alpha1.RAIDConfig) error {
+	if raidInterface == "no-raid" && raid != nil && (len(raid.HardwareRAIDVolumes) != 0 || len(raid.SoftwareRAIDVolumes) != 0) {
```

Comment:
Since we will [set the default value](https://github.com/metal3-io/baremetal-operator/pull/908/files/d3efbfbfce70825786f4b60fabd3490a974dfb4a#diff-a69d9029388ab766ed36b32180145f52785a9d4a153775510dbddfa928a72e1cR1127-R1138) for the empty raid field:
```go
{
		HardwareRAIDVolumes: nil or Status.Provisioning.RAID.HardwareRAIDVolumes(not empty),
		SoftwareRAIDVolume: [],
}
```
So if there is only ` raid= nil` will make mistakes.

Reply:
Fine, maybe a switch could be clearer here (minor point)

Reply:
Done.